stubs/dom.stub

```diff
@@ -34,6 +34,8 @@ class DOMNode
 
 class DOMElement extends DOMNode
 {
+	/** @var DOMNode|null */
+	public $firstChild;
```

Comment:
this makes the test green.. it leads to a few questions though:

- why does phpstan not pickup the already completed information from the phpstorm stubs?
- why does it not work when I declare the property on the `DOMNode` class, which would be correct per php-src manual

Reply:
ok.. I think I found that the php8-stubs and the jetbrains-stubs are only used atm for providing parameter and return types. the properties information from within class types seems not to be used currently

Reply:
if I read https://github.com/phpstan/phpstan-src/blob/master/src/PhpDoc/StubPhpDocProvider.php correctly, stub files reflect the runtime types and don't resolve concepts like inheritance.

thats the reason why it works when declaring the property at the `DOMElement` level but not at the `DOMNode` level.

Reply:
the way forward to resolve the underlying issue is to make phpstan aware of php8.1 stubs which contain the necessary reflection information, see https://github.com/phpstan/phpstan/issues/6213#issuecomment-997353206